src/acc/libsmm_acc/CMakeLists.txt

```diff
@@ -62,8 +62,10 @@ else () # i.e. USE_HIP
   set_source_files_properties(${LIBSMM_ACC_SRC_FILES} PROPERTIES HIP_SOURCE_PROPERTY_FORMAT 1)
   # hip_add_library does not support OBJECT libraries. For this reason, we are forced to use a static library
   hip_add_library(libsmm_acc STATIC ${LIBSMM_ACC_FILES})
+  target_include_directories(libsmm_acc PUBLIC ${HIP_PATH}/../include)
```

Comment:
does this have to be `PUBLIC`?

Reply:
I just tested with `PRIVATE` and `INTERFACE`, both of which fail. So, yes.

Reply:
Interesting, was it the examples/tests or dbcsr itself?

Reply:
the compilation fails in the tests, yes

Reply:
I see. For the tests we leak the CUDA/HIP headers through `libsmm_acc.h`, but as far as I can see that tests anyway bypass DBCSR and directly accesses `libsmm_acc`, hence the requirement for the CUDA/HIP headers.
For CUDA it seems we used the following in `tests/CMakeLists.txt` instead of exposing the include directory via a `PUBLIC` attribute:

```
      target_include_directories(
        ${libsmm_acc_test} PRIVATE ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
```

In the end, I am not sure whether it is worthwhile to follow this scheme to reduce the number of external include directories/libraries a consumer of the DBCSR library sees (what happens when we add a dependency with the `PUBLIC` attribute) or whether I am overly conscious about this stuff.

Reply:
No, I don't think you're being overly conscious, what you say makes a lot of sense. 
I just haven't been putting as much time and effort in the HIP backend's cmake, because HIP changes quite a bit with each new ROCm release, and still misses some cmake features important to us (e.g. support for object libraries). But I'll go ahead and fix what you pointed out, let's not let this get too messy ;)